pandas/io/parsers.py

```diff
@@ -1930,6 +1930,9 @@ def __init__(self, src, **kwds):
                 ]
             else:
                 self.names = list(range(self._reader.table_width))
+        else:
+            if self.prefix:
+                raise ValueError("argument prefix must be None if header not None")
```

Comment:
The error message is not 100% correct, since the header can be provided with `names`.